WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java

```diff
@@ -173,32 +172,8 @@ public static ReaderPost fromJson(JSONObject json) {
         // xpost info
         assignXpostIdsFromJson(post, json.optJSONArray("metadata"));
 
-        // if there's no featured image, check if featured media has been set - this is sometimes
-        // a YouTube or Vimeo video, in which case store it as the featured video so we can treat
-        // it as a video
-        if (!post.hasFeaturedImage()) {
-            JSONObject jsonMedia = json.optJSONObject("featured_media");
-            if (jsonMedia != null && jsonMedia.length() > 0) {
-                String mediaUrl = JSONUtils.getString(jsonMedia, "uri");
-                if (!TextUtils.isEmpty(mediaUrl)) {
-                    String type = JSONUtils.getString(jsonMedia, "type");
-                    boolean isVideo = (type != null && type.equals("video"));
-                    if (isVideo) {
-                        post.featuredVideo = mediaUrl;
-                    } else {
-                        post.featuredImage = mediaUrl;
-                    }
-                }
-            }
-        }
-        // if the post still doesn't have a featured image but we have attachment data, check whether
-        // we can find a suitable featured image from the attachments
-        if (!post.hasFeaturedImage() && post.hasAttachments()) {
-            post.featuredImage = new ImageSizeMap(post.getText(), post.attachmentsJson)
-                    .getLargestImageUrl(ReaderConstants.MIN_FEATURED_IMAGE_WIDTH);
-        }
-        // if we *still* don't have a featured image but the text contains an IMG tag, check whether
-        // we can find a suitable image from the text
+        // if the post doesn't have a featured image but it contains an IMG tag, check whether
+        // we can find a suitable image from the content
         if (!post.hasFeaturedImage() && post.hasImages()) {
             post.featuredImage = new ReaderImageScanner(post.text, post.isPrivate)
```

Comment:
> We've actually been choosing a featured image when there isn't one assigned for a long time

Sure thing - I'm questioning the decision of using `featuredImage` to hold on something _else_  than a featured image. Also it doesn't matter when the decision was made, we're looking at it and reconsidering now :)

I agree things are more attractive 100%, and I totally agree about the product / user point of view, the only thing I'm pointing out is the abuse of terminology in code: we are giving a class property a different meaning in logic than what the name describes or suggests. That's not clear, to say the least.

Now, the simplicity of the hack I think is what we are agreeing on, but we can't overlook the fact that we are using something for something else.

It might be clearer to use another property, like `coverImage`, and set and use that property wherever the layout needs a cover image, leaving `featuredImage` carry a value only when the Post actually does have a featured image.